TravelPoints-BackEnd/src/main/java/com/example/travelpointsbackend/service/implementation/WebSocketServiceImplementation.java

```diff
@@ -0,0 +1,52 @@
+package com.example.travelpointsbackend.service.implementation;
+
+import com.example.travelpointsbackend.dto.SocketMessageDTO;
+import com.example.travelpointsbackend.dto.TouristAttractionDetailDTO;
+import com.example.travelpointsbackend.model.TouristAttraction;
+import com.example.travelpointsbackend.model.User;
+import com.example.travelpointsbackend.model.UserWishlist;
+import com.example.travelpointsbackend.repository.TouristAttractionRepository;
+import com.example.travelpointsbackend.repository.UserWishlistRepository;
+import com.example.travelpointsbackend.service.WebSocketService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class WebSocketServiceImplementation implements WebSocketService {
+    @Autowired
+    private SimpMessagingTemplate simpMessagingTemplate;
+    private final UserWishlistRepository userWishlistRepository;
+    private final TouristAttractionRepository touristAttractionRepository;
+
+    // nu stiu daca e okay
+    @Override
+    public void sendMessage(Long attractionId, TouristAttractionDetailDTO touristAttractionDetailDTO){
+        System.out.println("Am intrat in send!");
+        Optional<TouristAttraction> touristAttractionOld = touristAttractionRepository.findById(attractionId);
+        if(touristAttractionOld.isPresent()){
+            if (!touristAttractionOld.get().getAttractionDetail().getOffers().equals(touristAttractionDetailDTO.getOffers())){
+                List<UserWishlist> userWishlists = userWishlistRepository.findAll();
+                for(UserWishlist userWishlist : userWishlists) {
+                    List<TouristAttraction> touristAttractions = userWishlist.getTouristAttractionList();
+                    if(touristAttractions.contains(touristAttractionOld.get())){
+                        User user = userWishlist.getUser();
+                        if(user.isLogged()){
+                            Long userId = user.getUserId();
+                            String attractionName = touristAttractionDetailDTO.getName();
+                            String offers = touristAttractionDetailDTO.getOffers();
+                            SocketMessageDTO socketMessageDTO = new SocketMessageDTO(userId, attractionName, offers);
+                            System.out.println("Ne aflam inainte de websocket!");
+                            simpMessagingTemplate.convertAndSend("/topic/" + userId, socketMessageDTO);
```

Comment:
this method should only contain this line and it should be called only when the offer is updated with an offer. basically, move lines 31-44 to TouristAttractionController and to this method send the user ids and the dto